pegasus/helpers/section_api_helpers.rb

```diff
@@ -21,23 +21,32 @@ def self.fetch_user_students(user_id)
   def self.create(params)
     name = !params[:name].to_s.empty? ? params[:name].to_s : 'New Student'
     gender = valid_gender?(params[:gender]) ? params[:gender] : nil
+    provider = supported_provider?(params[:provider]) ? params[:provider] : 'sponsored'
     birthday = age_to_birthday(params[:age]) ?
       age_to_birthday(params[:age]) : params[:birthday]
 
     created_at = DateTime.now
 
-    row = Dashboard.db[:users].insert(
+    data =
       {
         name: name,
         user_type: 'student',
-        provider: 'sponsored',
+        provider: provider,
         gender: gender,
         birthday: birthday,
         created_at: created_at,
         updated_at: created_at,
         username: UserHelpers.generate_username(Dashboard.db[:users], name)
       }.merge(random_secrets)
-    )
+    if provider == 'sponsored'
+      row = Dashboard.db[:users].insert(data)
+    else
+      uid = params[:uid].to_s
```

Comment:
Following the pattern on [line 22](https://github.com/code-dot-org/code-dot-org/blob/19896508a041b645414e385c7b651a8cc401af97/pegasus/helpers/section_api_helpers.rb#L22).  I'm not super familiar with the Sequel gem, so exercising an abundance of caution to prevent any SQL injection 😛.